lib/Doctrine/ConnectionFactory.php

```diff
@@ -49,7 +50,11 @@ public function createConnection() {
       $connectionParams['unix_socket'] = Env::$dbSocket;
     } else {
       $connectionParams['host'] = Env::$dbIsIpv6 ? ('[' . Env::$dbHost . ']') : Env::$dbHost;
-      $connectionParams['port'] = Env::$dbPort;
+      if (!empty(Env::$dbPort)) {
+        $connectionParams['port'] = Env::$dbPort;
+      } else {
+        $connectionParams['port'] = $wpdb->get_var('SELECT @@port');
```

Comment:
You removed two checks on the port in the integration tests. It would be nice to have a test for this new option. Do you think that it is possible to add it? Maybe I'm wrong. It is only an idea.

Reply:
I added a test when using `Env::$dbPort` but I am not sure how to test this line.